docs/DeveloperGuide.md

```diff
@@ -63,15 +64,105 @@ In this section, you will learn about the general design and structure TrackIt@N
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+[**`Commons`**](#common-classes) represents a collection of classes used by multiple other components.
+
+The rest of the App consists of four components.
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Logic`**](#logic-component): The command executor & the logic
+  interface providing APIs for the UI to retrieve necessary data.
+* [**`Model`**](#model-component): Holds the data of the App in memory.
+* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+
+Each of the four components,
+
+* defines its *API* in an `interface` with the same name as the Component.
+* exposes its functionality using a concrete `{Component Name}Manager` class (which implements the corresponding API `interface` mentioned in the previous point.
+
+For example, the `Logic` component (see the class diagram given below) defines its API in the `Logic.java` interface and exposes its functionality using the `LogicManager.java` class which implements the `Logic` interface.
+
+![Class Diagram of the Logic Component](images/LogicClassDiagram.png)
+
+**How the architecture components interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues
+ the command `T delete 1`.
+
+<img src="images/ArchitectureDeleteTaskSequenceDiagram.png" width="587" />
+
+Another *Sequence Diagram* below shows how the components interact with each other for the scenario where the user
+ click on a module tab.
+
+<img src="images/ArchitectureModuleTabSequenceDiagram.png" width="686" />
+
+The sections below give more details of each component.
+
 
 ### **UI Component** <a name="ui"></a>
 
 ### **Logic Component** <a name="logic"></a>
 
+![Structure of the Logic Component](images/LogicClassDiagram.png)
+
+**API** :
+[`Logic.java`](https://github.com/AY2021S1-CS2103T-W13-4/tp/blob/master/src/main/java/trackitnus/logic/Logic.java)
+
+1. `Logic` uses the `TrackIterParser` class to parse the user command.
+1. This results in a `Command` object which is executed by the `LogicManager`.
+1. The command execution can affect the `Model` (e.g. adding a person).
+1. The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
+1. In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+
+![Interactions Inside the Logic Component for the `delete 1` Command](images/LogicDeleteTaskSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `DeleteTaskCommandParser
+` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
```

Comment:
Somehow the **Note** is not correctly displayed. But since we already have the information source I think we can just remove the **Note** completely. Also, change last line to `due to a limitation in PlantUML, the lifeline continues to the end of the diagram`